src/main/java/bio/terra/service/dataset/Dataset.java

```diff
@@ -132,12 +132,14 @@ public void validateDatasetAssetSpecification(AssetModel assetModel) {
 
     // Follow should reference an existing relationship as defined in the original dataset create
     // query
-    for (var assetFollow : assetModel.getFollow()) {
-      if (!relationships.stream().anyMatch(r -> r.getName().equals(assetFollow))) {
-        errors.add(
-            "Relationship specified in follow list '"
-                + assetFollow
-                + "' does not exist in dataset's list of relationships");
+    if (assetModel.getFollow() != null) {
+      for (var assetFollow : assetModel.getFollow()) {
+        if (!relationships.stream().anyMatch(r -> r.getName().equals(assetFollow))) {
+          errors.add(
+              "Relationship specified in follow list '"
+                  + assetFollow
+                  + "' does not exist in dataset's list of relationships");
+        }
```

Comment:
@pshapiro4broad  - Is there a better way to perform this null check before iterating?

Reply:
Oh nice! I like that better. thanks!

Reply:
Guess I missed this; personally I'd use `!= null` over `emptyIfNull()` in this case, but either works.

Did you consider changing the openapi API spec to specify a default value? If you could do that, then you wouldn't need to check for `null` here.